rewriter/RequiresAncestorKernel.cc

```diff
@@ -0,0 +1,59 @@
+#include "rewriter/RequiresAncestorKernel.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "core/Names.h"
+#include "core/core.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+namespace {
+
+bool hasAnyRequiresAncestorCall(core::MutableContext ctx, ast::ClassDef *klass) {
+    for (auto &stat : klass->rhs) {
+        if (auto send = ast::cast_tree<ast::Send>(stat)) {
+            if (send->fun == core::Names::requiresAncestor() && send->recv.isSelfReference() && send->hasBlock()) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+} // namespace
+
+void RequiresAncestorKernel::run(core::MutableContext ctx, ast::ClassDef *klass) {
+    // Only run for Module definitions
```

Comment:
I would drop these first 3 comments, they're pretty verbatim to what the code already says